redis/redis.go

```diff
@@ -46,7 +71,7 @@ func (b *bucket) updateOldReset() error {
 	if err != nil {
 		return err
 	}
-	b.reset = time.Now().Add(time.Duration(ttl.(int64) * millisecond))
+	b.setReset(time.Now().Add(time.Duration(ttl.(int64) * millisecond)))
```

Comment:
Feels strange to read `Reset()` above, but then lock the mutex here in `setReset`